src/object/cli_obj.c

```diff
@@ -2215,7 +2215,8 @@ dc_obj_query_key(tse_task_t *api_task)
 	if (rc) {
 		if (rc != -DER_INVAL)
 			goto out_task;
-		epoch = daos_ts2epoch();
+		/* FIXME: until distributed transaction. */
+		epoch = DAOS_EPOCH_MAX; /* = daos_ts2epoch();*/
```

Comment:
seems to me that should assign the HLC for key query at client-side, if the req will send to multiple shards at client-side.

but assign at client-side will with HLC out-of-sync problem with server-side modification HLC, need a kind of hlc sync mechanism. So before that is ready it looks fine to me to land this patch first.